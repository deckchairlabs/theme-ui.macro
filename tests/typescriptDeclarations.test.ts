import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

pluginTester({
  plugin,
  pluginName: 'theme-ui.macro',
  title: 'TypeScript Declaration',
  snapshot: false,
  babelOptions: {
    presets: ['@babel/preset-typescript'],
    filename: __filename,
  },
  pluginOptions: {
    themeUI: {
      generateTSDeclaration: {
        output: path.join(__dirname, 'generated/theme.d.ts'),
      },
    },
  },
  fixtures: path.join(__dirname, 'plugins/typescriptDeclarations'),
})
