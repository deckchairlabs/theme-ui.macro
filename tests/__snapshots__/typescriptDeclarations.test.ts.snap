// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generated declaration file matches snapshot 1`] = `
declare module "@theme-ui/css" {
    export interface Theme {
        colors: {
            primary: "red";
            secondary: "blue";
            black: "#000";
            highlight: "colors.primary";
        };
        space: [
            0,
            4,
            8,
            16
        ];
        buttons: {
            base: {
                padding: [
                    2,
                    3
                ];
            };
        };
    }
}
export *;

`;
